qt/scientific_interfaces/Indirect/test/IndirectFitDataTest.h

```diff
@@ -0,0 +1,447 @@
+      numberOfSpectra, numberOfBins);
+  Spectra const spec = std::make_pair(0u, workspace->getNumberHistograms() - 1);
+  IndirectFitData data(workspace, spec);
+  return data;
+}
+
+/// Simple class to set up the ADS with the configuration required
+struct SetUpADSWithWorkspace {
+
+  SetUpADSWithWorkspace(std::string const &inputWSName,
+                        IndirectFitData const &data) {
+    AnalysisDataService::Instance().addOrReplace(inputWSName, data.workspace());
+  };
+
+  ~SetUpADSWithWorkspace() { AnalysisDataService::Instance().clear(); };
+};
+
+/// This is used to compare Spectra which is implemented as a boost::variant
+struct AreSpectraEqual : public boost::static_visitor<bool> {
+
+  template <typename T, typename U>
+  bool operator()(const T &, const U &) const {
+    return false; // cannot compare different types
+  }
+
+  template <typename T> bool operator()(const T &lhs, const T &rhs) const {
+    return lhs == rhs;
+  }
+};
+
+} // namespace
+
+class IndirectFitDataTest : public CxxTest::TestSuite {
+public:
+  static IndirectFitDataTest *createSuite() {
+    return new IndirectFitDataTest();
+  }
+
+  static void destroySuite(IndirectFitDataTest *suite) { delete suite; }
+
+  void test_data_is_instantiated_correctly() {
+    auto const workspace = WorkspaceCreationHelper::create2DWorkspace123(1, 3);
+    Spectra const spec =
+        std::make_pair(0u, workspace->getNumberHistograms() - 1);
+
+    workspace->setTitle("Test Title");
+    IndirectFitData const data(workspace, spec);
+
+    TS_ASSERT_EQUALS(data.workspace(), workspace);
+    TS_ASSERT_EQUALS(data.workspace()->getTitle(), "Test Title");
+    TS_ASSERT_EQUALS(data.workspace()->getNumberHistograms(), 1);
+  }
+
+  void test_that_DiscontinuousSpectra_is_set_correctly() {
+    DiscontinuousSpectra<std::size_t> const spectra =
+        DiscontinuousSpectra<std::size_t>("0-5,8,10");
+
+    std::string const spectraString = "0-5,8,10";
+    std::vector<std::size_t> const spectraVec{0, 1, 2, 3, 4, 5, 8, 10};
+
+    TS_ASSERT_EQUALS(spectra.getString(), spectraString);
+    for (auto it = spectra.begin(); it < spectra.end(); ++it)
+      TS_ASSERT_EQUALS(*it, spectraVec[it - spectra.begin()]);
+  }
+
+  void test_data_is_stored_correctly_in_the_ADS() {
+    auto const data = getIndirectFitData(1, 3);
+    SetUpADSWithWorkspace ads("WorkspaceName", data);
+
+    TS_ASSERT(AnalysisDataService::Instance().doesExist("WorkspaceName"));
+    MatrixWorkspace_sptr workspace =
+        boost::dynamic_pointer_cast<MatrixWorkspace>(
+            AnalysisDataService::Instance().retrieve("WorkspaceName"));
+    TS_ASSERT_EQUALS(workspace->getNumberHistograms(), 1);
+  }
+
+  void test_displayName_returns_correct_name() {
+    auto const data = getIndirectFitData(1, 3);
+
+    std::vector<std::string> const formatStrings{
+        "%1%_s%2%_Result", "%1%_f%2%,s%2%_Parameter", "%1%_s%2%_Parameter"};
+    std::string const rangeDelimiter = "_to_";
```

Comment:
Where does the `rangeDelimeter` actually get used in this test? Should I be expecting to see it in the name somewhere?

Reply:
It is used in the ASSERT statements. Yea, I think the name could be better